src/current/v25.2/create-policy.md

```diff
@@ -0,0 +1,113 @@
+
+| Command / clause pattern            | `SELECT` policy - `USING` (row that already exists) | `INSERT` policy - `WITH CHECK` (row being added) | `UPDATE` policy - `USING` (row before the change) | `UPDATE` policy - `WITH CHECK` (row after the change) | `DELETE` policy - `USING` (row to be removed) |
+|-------------------------------------|-----------------------------------------------------|--------------------------------------------------|---------------------------------------------------|-------------------------------------------------------|-----------------------------------------------|
+| `SELECT`                            | ✓                                                   | —                                                | —                                                 | —                                                     | —                                             |
+| `SELECT ... FOR UPDATE / FOR SHARE` | ✓                                                   | —                                                | ✓                                                 | —                                                     | —                                             |
+| `INSERT / MERGE ... THEN INSERT`    | —                                                   | ✓                                                | —                                                 | —                                                     | —                                             |
+| `INSERT ... RETURNING`              | ✓ ‡                                                 | ✓                                                | —                                                 | —                                                     | —                                             |
+| `UPDATE / MERGE ... THEN UPDATE`    | ✓ ‡                                                 | —                                                | ✓                                                 | ✓                                                     | —                                             |
+| `DELETE`                            | ✓ ‡                                                 | —                                                | —                                                 | —                                                     | ✓                                             |
+| `INSERT ... ON CONFLICT DO UPDATE`  | ✓                                                   | —                                                | ✓                                                 | ✓                                                     | —                                             |
+
+‡ The `SELECT` check is only evaluated when the statement actually needs to read from the relation, e.g., in a `WHERE`, `SET`, or `RETURNING` clause that references table columns.
+
+CockroachDB's behavior differs from that described in the table above in the following ways:
```

Comment:
I like the chart that we have above. But nowhere does it say it's the postgres output. I would prefer to keep the chart CRDB specific. Could we phrase the postgres vs crdb behaviour difference in another way?

Reply:
i'm definitely endeavoring to make the chart CRDB-specific. It's not intended to be the PG output, but our own. 

agree the wording makes that unclear, sorry

i have reworded things a bit to remove the "Vs PG" type language, PTAL

> Additional considerations include:

> - `SELECT` evaluation: CockroachDB always evaluates `SELECT` (`USING`) policies for `INSERT` and `UPDATE`, even when the statement doesn't reference table columns.
> - `ON CONFLICT ... DO NOTHING`: CockroachDB still runs constraint and row-level policy checks on the `VALUES` clause even when the candidate row is discarded because of a conflict. This is a known limitation described in [cockroachdb/cockroach#35370](https://github.com/cockroachdb/cockroach/issues/35370).